File: js/arrays_strings/boat_to_save_people.js
const requiredBoatToSavePeople = (people, limit) => {
    people.sort(function(a, b){return a-b});
    l = 0;
    h = people.length - 1;
    let boat = 0;
    while(l <= h){
        if(people[l] + people[h] <= limit){
            l++;
            h--;
        } else{
            h--;
        }
        boat++;
    }
    return boat;
}
const limit = 50;
const people = [44,10,29,12,49,41,23,5,17,26];
console.log(requiredBoatToSavePeople(people, limit))

File: python/arrays_strings/boat_to_save_people.py
from typing import List

class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        boats = 0
        left = 0
        right = len(people) -1
        while ( left <= right):
            if(people[left] + people[right] <= limit):
                left+= 1
                right-=1
            else:
                right-=1           
            boats+=1
        return boats

s = Solution()
people = [3,2,2,1]
limit = 3
ans = s.numRescueBoats(people, limit)
print(ans)